test/integration/appfactory/appfactory_test.go

```diff
@@ -15,23 +15,135 @@
+
+		// check app infra repo
+		repositoryName := appFactory.GetStringOutput("app_infra_repository_name")
+		repoURL := fmt.Sprintf("https://source.developers.google.com/p/%s/r/%s", adminProjectID, repositoryName)
+		repoOP := gcloud.Runf(t, "source repos describe %s --project %s", repositoryName, adminProjectID)
+		assert.Equal(repoURL, repoOP.Get("url").String(), "source repo %s should have url %s", repositoryName, repoURL)
+
+		// check workspace SA access to repo
+		repoSa := fmt.Sprintf("serviceAccount:tf-cb-%s@%s.iam.gserviceaccount.com", repositoryName, adminProjectID)
+		repoIamOpts := gcloud.WithCommonArgs([]string{"--flatten", "bindings", "--filter", "bindings.role:roles/viewer", "--format", "json"})
+		repoIamPolicyOp := gcloud.Run(t, fmt.Sprintf("source repos get-iam-policy %s --project %s", repositoryName, adminProjectID), repoIamOpts).Array()[0]
+		listMembers := utils.GetResultStrSlice(repoIamPolicyOp.Get("bindings.members").Array())
+		assert.Contains(listMembers, repoSa, fmt.Sprintf("Service Account %s should have role roles/viewer on repo %s", repoSa, repositoryName))
+
+		// check cloudbuild workspace
+		// buckets
+		gcloudArgsBucket := gcloud.WithCommonArgs([]string{"--project", adminProjectID, "--json"})
+		for _, bucket := range []struct {
+			output string
+			suffix string
+		}{
+			{
+				output: "app_cloudbuild_workspace_state_bucket_name",
+				suffix: "state",
+			},
+			{
+				output: "app_cloudbuild_workspace_logs_bucket_name",
+				suffix: "logs",
+			},
+			{
+				output: "app_cloudbuild_workspace_artifacts_bucket_name",
+				suffix: "build",
+			},
+		} {
+			bucketSelfLink := appFactory.GetStringOutput(bucket.output)
+			opBucket := gcloud.Run(t, fmt.Sprintf("storage ls --buckets gs://bkt-%s-app1-%s", adminProjectID, bucket.suffix), gcloudArgsBucket).Array()
+			assert.Equal(bucketSelfLink, opBucket[0].Get("metadata.selfLink").String(), fmt.Sprintf("The bucket SelfLink should be %s.", bucketSelfLink))
+		}
+		// triggers
+		repoName := appFactory.GetStringOutput("app_infra_repository_name")
+		for _, trigger := range []struct {
+			output string
+			file   string
+		}{
+			{
+				output: "app_cloudbuild_workspace_apply_trigger_id",
+				file:   "cloudbuild-tf-apply.yaml",
+			},
+			{
+				output: "app_cloudbuild_workspace_plan_trigger_id",
+				file:   "cloudbuild-tf-plan.yaml",
+			},
+		} {
+			triggerID := testutils.GetLastSplitElement(appFactory.GetStringOutput(trigger.output), "/")
+			buildTrigger := gcloud.Runf(t, "builds triggers describe %s --project %s --region %s", triggerID, adminProjectID, "global")
+			filename := buildTrigger.Get("filename").String()
+			assert.Equal(trigger.file, filename, fmt.Sprintf("The filename for the trigger should be %s but got %s.", trigger.file, filename))
+			assert.Equal(repoName, buildTrigger.Get("triggerTemplate.repoName").String(), "the trigger should use the repo %s", repoName)
+		}
+
+		// check env projects
+		cloudBuildSARoles := []string{"roles/owner"}
```

Comment:
I know the `roles/owner` is part of the code. I was just wondering if it could be changed for another role with least privileges. Anyway, it isn`t something should affect the test. Just sharing some toughts.

Reply:
I think it should be changed to a list of more granular roles, but I think we will need more code to find out which are these roles